tutorials/chapter1_annexa.md

```diff
@@ -0,0 +1,299 @@
+    }
+
+    std::function<void(const T &)> forwarderListenerFunction;
+};
+
+void listener(const std::string &message)
+{
+    std::cout << "Message received in free function \"listener\": " << message << std::endl;
+    std::cout << "Provider (subscriber) thread ID: " << std::this_thread::get_id() << std::endl;
+}
+```
+
+### Example 1: Passing a free function
+
+The easiest form to pass a function is passing a free function. In this case the ```listener``` free function from ```simple_listener.h``` has been used:
+
+```cpp
+#include "simple_listener.h"
+#include "simple_publisher.h"
+
+#include <iostream>
+#include <klepsydra/core/event_emitter_middleware_provider.h>
+
+int main()
+{
+    std::cout << "Main thread ID: " << std::this_thread::get_id() << std::endl;
+
+    kpsr::EventEmitterMiddlewareProvider<std::string> provider(nullptr,
+                                                               "tutorial_app_api_example1",
+                                                               0,
+                                                               nullptr,
+                                                               nullptr);
+
+    SimplePublisher simplePublisher(provider.getPublisher());
+
+    provider.getSubscriber()->registerListener("example1", listener);
+
+    simplePublisher.run();
+
+    provider.getSubscriber()->removeListener("example1");
+}
+```
+
+### Example 2: Passing a class member function
+
+A class member function can be passed as the listener function too. In this case, the ```staticListener``` method from the ```SimpleListener``` class has been used:
+
+```cpp
+#include "simple_listener.h"
+#include "simple_publisher.h"
+
+#include <iostream>
+#include <klepsydra/core/event_emitter_middleware_provider.h>
+
+int main()
+{
+    std::cout << "Main thread ID: " << std::this_thread::get_id() << std::endl;
+
+    kpsr::EventEmitterMiddlewareProvider<std::string> provider(nullptr,
+                                                               "tutorial_app_api_example2",
+                                                               0,
+                                                               nullptr,
+                                                               nullptr);
+
+    SimplePublisher simplePublisher(provider.getPublisher());
+
+    provider.getSubscriber()->registerListener("example2",
+                                               SimpleListener<std::string>::staticListener);
+
+    simplePublisher.run();
+
+    provider.getSubscriber()->removeListener("example2");
+}
+```
+
+## Example 3: Passing a class function member which is bound to another one
+
+Functions can be bound to another ones in cases where, for example, the functionality will be the same but the number of arguments differ. In this case, the ```forwarderListenerFunction``` function from the ```SimpleListener``` class has been used:
```

Comment:
In this example `listener` still has only 1 arguments. So it might be helpful to define `void listenerWithMultipleArgs(const T& event, const std::string &name)` to illustrate the point you are making here.